gcsio/src/main/java/com/google/cloud/hadoop/gcsio/testing/InMemoryGoogleCloudStorage.java

```diff
@@ -547,6 +547,7 @@ public GoogleCloudStorageItemInfo composeObjects(
       final StorageResourceId destination,
       CreateObjectOptions options)
       throws IOException {
+    Preconditions.checkArgument(sources.size() <= 32, "Cannot compose more than 32 sources");
```

Comment:
Use the same constant so that it can be updated in one place. Alternatively, we can remove this Precondition.

Reply:
I moved the constant to `GoogleCloudStorage` to do this. (Also suggested by @medb.)